### PR TITLE
Lighting Balance Pass

### PR DESCRIPTION
**Pull request by NanakoAC**
Tweaks diona values on all lights, and range/power on a couple

Also adds a slightly dimmer sprite for tube light fixtures that they switch to during night mode. To reduce the incongruity from having a bright tube putting out very little light

### CODE REVIEW

File: code/game/objects/items/devices/flashlight.dm [hunk 1]
```diff
@@ -137,10 +140,11 @@
 	icon_state = "lamp"
 	item_state = "lamp"
 	brightness_on = 5
-	w_class = 4
+	w_class = 5
```
Comment (NanakoAC) [inline code/game/objects/items/devices/flashlight.dm]: this change makes lamps too large to fit in backpacks, that's deliberate

File: code/modules/lighting/lighting_atom.dm [hunk 2]
```diff
@@ -4,7 +4,8 @@
 	var/light_power = 1 // Intensity of the light.
 	var/light_range = 0 // Range in tiles of the light.
 	var/light_color     // Hexadecimal RGB string representing the colour of the light.
-	var/uv_intensity	// How much UV light is being emitted by this object. Valid range: 0-255.
+	var/uv_intensity = 255	// How much UV light is being emitted by this object. Valid range: 0-255.
```
Comment (Lohikar) [inline code/modules/lighting/lighting_atom.dm]: No.

File: code/modules/mob/living/silicon/silicon.dm [hunk 3]
```diff
@@ -24,7 +24,7 @@
 	var/next_alarm_notice
 	var/list/datum/alarm/queued_alarms = new()
 
-
+	uv_intensity = 175 //Lights cast by robots have reduced effect on diona
```
Comment (Lohikar) [inline code/modules/mob/living/silicon/silicon.dm]: Why would they have any? Why would a robot have a presumably more expensive UV-emissive light?

File: code/modules/detectivework/tools/uvlight.dm [hunk 4]
```diff
@@ -11,15 +11,15 @@
...
+	uv_intensity = 255
 	var/range = 3
 	var/on = 0
 	var/step_alpha = 50
 
 /obj/item/device/uv_light/attack_self(var/mob/user)
 	on = !on
 	if(on)
-		set_light(range, 2, "#007fff", uv = 100)
+		set_light(range, 2, "#7700dd", uv = uv_intensity)
```
Comment (NanakoAC) [inline code/modules/detectivework/tools/uvlight.dm]: This color is a correct match to the purple shade of the sprite

File: code/modules/mob/living/carbon/diona_base.dm [hunk 5]
```diff
@@ -415,21 +415,22 @@ var/list/diona_banned_languages = list(
...
 		else if (DS.light_organ.is_bruised())
-			light_factor = 0.8
+			light_factor *= 0.8
 	else if (DS.dionatype == 2)
-		light_factor = 0.8
+		light_factor = 1
 
 	if (T)
-		var/raw = T.get_uv_lumcount(0, 2) * light_factor * 5.5
+		var/raw = min(T.get_uv_lumcount(0, 2) * light_factor * 5.5, 5.5)
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/diona_base.dm]: `get_uv_lumcount` can only return a value between 0 and 1, this is unnecessary.

File: code/modules/mob/living/carbon/diona_base.dm [hunk 6]
```diff
@@ -415,21 +415,22 @@ var/list/diona_banned_languages = list(
 //GETTER FUNCTIONS
 
 /mob/living/carbon/proc/get_lightlevel_diona(var/datum/dionastats/DS)
-	var/light_factor = 1
+	var/light_factor = 1.15
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/diona_base.dm]: diona light intake levels have been buffed a bit. All the lights were balanced around this new multiplier

File: code/game/objects/items/devices/flashlight.dm [hunk 7]
```diff
@@ -252,6 +258,7 @@
 	icon_state = "glowstick"
 	item_state = "glowstick"
 	contained_sprite = 1
+	uv_intensity = 255
```
Comment (Lohikar) [inline code/game/objects/items/devices/flashlight.dm]: Why would a glowstick emit this much UV?

File: code/modules/mob/living/carbon/diona_base.dm (hunk 5 above)
Comment (NanakoAC) [inline code/modules/mob/living/carbon/diona_base.dm]: min to prevent this going over the limit

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 8]
```diff
@@ -12,8 +12,7 @@ var/global/list/obj/item/device/pda/PDAs = list()
 	w_class = 2.0
 	slot_flags = SLOT_ID | SLOT_BELT
 	sprite_sheets = list("Resomi" = 'icons/mob/species/resomi/id.dmi')
-	//offset_light = 1
-	//diona_restricted_light = 1//Light emitted by this object or creature has limited interaction with diona
+	uv_intensity = 15
```
Comment (Lohikar) [inline code/game/objects/items/devices/PDA/PDA.dm]: I would like to get Hurtz' and @skull132's opinion on PDAs emitting *any* UV light.

File: code/modules/mob/living/carbon/diona_base.dm (hunk 5 above)
Comment (Lohikar) [inline code/modules/mob/living/carbon/diona_base.dm]: Never mind, didn't take `light_factor` into account.

File: code/modules/lighting/lighting_atom.dm (hunk 2 above)
Comment (NanakoAC) [inline code/modules/lighting/lighting_atom.dm]: Rudeness won't help anyone, please remain civil

**Comment by NanakoAC** [comment]
@Lohikar 
It is not UV. Don't think of it as UV. Don't call it UV. If it'll make you feel better I can comb through and change the variable names.

It is a generic diona scalar, and its used as a fudge factor to compensate for several variables, including: Light power, light range, distance from light, accessibility of the source, and limited resources (fuel/battery/etc), and yes, also scientific believability. As can be seen the UV light has a higher UV value than most other portable lights, and space still short circuits to maximum due to starlight. UV is already represented and the entire system does not need to be devoted to it.

These values are not meaningful in isolation, you're not going to mentally account for the aforementioned variables. Looking at a glowstick and complaining about its UV value is nonsensical.  A glowstick at 255 UV helps a diona far less than a flare at 100 UV

The values here, like the original values, are based on a top-down design of balancing to produce expected results. This design was originally made in tandem with killerhurtz and he signed off on it. All values were calibrated to produce specific survival times. And all new values have been recalibrated with the same goals in mind

All of my 'changes' in this PR, with the exception of the tubelight sprite, are not. They are simply restoring diona balance to almost its original state, which nobody ever complained about. If killerhurtz wishes to change things i can do so, and ill have a talk with him to gauge any differences in expected results, and adjust values.

I told you i'd do a balance pass on this because i'm the diona maintainer and making them fit properly in the new system is my responsibility to handle. You don't need to worry about it now. You've made your new lighting system and its good, your work here is done.

**Comment by NanakoAC** [comment]
I've explained the state of balance to killerhurtz and he approves of this

**Comment by skull132** [comment]
@Lohikar Is there a chance of us running into an issue whereby dried up blood that's sprayed with luminol now appears in hallways? Or we good on that front?

Also, hindsight is a bitch but it must be bestowed regardless: _all of this could have been avoided if we simply knew that negating stackable light sources was as simple as picking the best light from the mob as the light source to use. :silent-ree:_

**Comment by Lohikar** [comment]
@skull132 The mechanics of luminol haven't been changed, UV light isn't actually being treated as UV due to game balance reasons.

As nanako said, it's not really UV, just a generic diona balancing value.

It probably should be renamed.

File: code/game/objects/items/devices/flashlight.dm (hunk 1 above)
Comment (Lohikar) [inline code/game/objects/items/devices/flashlight.dm]: Why does this need changing?

File: code/game/objects/items/devices/flashlight.dm [hunk 9]
```diff
@@ -165,13 +169,14 @@
 	name = "flare"
 	desc = "A red standard-issue flare. There are instructions on the side reading 'pull cord, make light'."
 	w_class = 2.0
-	brightness_on = 8 // Pretty bright.
-	light_power = 3
+	brightness_on = 4 // Pretty bright.
+	light_power = 4
```
Comment (Lohikar) [inline code/game/objects/items/devices/flashlight.dm]: Light power should generally be 1 or under, can you post a screenshot of what this looks like?

File: code/game/objects/items/devices/flashlight.dm (hunk 1 above)
Comment (skull132) [inline code/game/objects/items/devices/flashlight.dm]: It used to be like that. But new Bay storage system fucked some other objects like this as well. IIRC.

**Comment by skull132** [comment]
Uh, can you please post before/afters on the light power and intensity adjustments? Lohikar made well sure to consult #developers with any and all changes, so those have all been reviewed and approved.

File: code/game/objects/items/devices/flashlight.dm (hunk 1 above)
Comment (NanakoAC) [inline code/game/objects/items/devices/flashlight.dm]: This change is so that diona don't steal lamps and carry them around all day

**Comment by skull132** [comment]
@NanakoAC plz. Pictures.

So this could be merged.

File: code/game/objects/items/devices/flashlight.dm (hunk 9 above)
Comment (NanakoAC) [inline code/game/objects/items/devices/flashlight.dm]: We discussed this before back when we were testing it together. Powers above 1 falloff less and stay brighter for longer

**Comment by NanakoAC** [comment]
RE: Power and intensity. I changed one and only one light source, flares.

Flares BEFORE my changes


Insanely bright, illuminates almost the whole screen fully. I felt this was dumb and inconsistent with how flares are supposed to work.

Flares AFTER my changes


Much smaller area, still brightly lit.
I think lohikar was sort of right in that the power isnt relevant here, the change from 3 to 4 power does nothing in this case. reduction of range is all that matters. flares are a fairly localised lightsource for marking things.

**Pull request merged**